core/src/main/java/com/linecorp/armeria/common/HttpResponse.java

```diff
@@ -419,6 +422,79 @@ static HttpResponse of(ResponseHeaders headers, Publisher<? extends HttpObject>
         return PublisherBasedHttpResponse.from(headers, publisher);
     }
 
+    /**
+     * Creates a new HTTP response with the specified {@code content} that is converted into JSON using the
+     * default {@link ObjectMapper}.
+     *
+     * @throws IllegalArgumentException if failed to encode the {@code content} into JSON.
+     * @see JacksonModuleProvider
+     */
+    static HttpResponse ofJson(Object content) {
+        return ofJson(HttpStatus.OK, content);
+    }
+
+    /**
+     * Creates a new HTTP response with the specified {@link HttpStatus} and {@code content} that is
+     * converted into JSON using the default {@link ObjectMapper}.
+     *
+     * @throws IllegalArgumentException if failed to encode the {@code content} into JSON.
+     * @see JacksonModuleProvider
+     */
+    static HttpResponse ofJson(HttpStatus status, Object content) {
+        requireNonNull(status, "status");
+        final ResponseHeaders headers = ResponseHeaders.builder(status)
+                                                       .contentType(MediaType.JSON)
+                                                       .build();
+        return ofJson(headers, content);
+    }
+
+    /**
+     * Creates a new HTTP response with the specified {@link MediaType} and {@code content} that is
+     * converted into JSON using the default {@link ObjectMapper}.
+     *
+     * @throws IllegalArgumentException if the specified {@link MediaType} is not a JSON compatible type; or
+     *                                  if failed to encode the {@code content} into JSON.
+     * @see JacksonModuleProvider
+     */
+    static HttpResponse ofJson(MediaType contentType, Object content) {
+        requireNonNull(contentType, "contentType");
+        checkArgument(contentType.isJson(),
+                      "contentType: %s (expected: the subtype is 'json' or ends with '+json'.");
+        final ResponseHeaders headers = ResponseHeaders.builder(HttpStatus.OK)
+                                                       .contentType(contentType)
+                                                       .build();
+        return ofJson(headers, content);
+    }
+
+    /**
+     * Creates a new HTTP response with the specified {@link ResponseHeaders} and {@code content} that is
+     * converted into JSON using the default {@link ObjectMapper}.
+     *
+     * @throws IllegalArgumentException if failed to encode the {@code content} into JSON.
+     * @see JacksonModuleProvider
+     */
+    static HttpResponse ofJson(ResponseHeaders headers, Object content) {
+        requireNonNull(headers, "headers");
+        requireNonNull(content, "content");
+
+        final HttpData httpData;
+        try {
+            httpData = HttpData.wrap(JacksonUtil.getDefaultObjectMapper().writeValueAsBytes(content));
+        } catch (JsonProcessingException e) {
+            throw new IllegalArgumentException(e);
```

Comment:
How about including a message so a user doesn't have to look down in the cause?
```suggestion
            throw new IllegalArgumentException(e.toString(), e);
```